tests-cpp/test_1560-builder-options.cpp

```diff
@@ -0,0 +1,14 @@
+// BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+#include "awkward/BuilderOptions.h"
+#include <cassert>
+
+int main(int /* argc */, char ** /* argv */) {
+
+  awkward::Options<int64_t, double> options(1024, 1.0);
```

Comment:
For a C++ user, this should have a name (or be in a namespace) that indicates that this object represents options for builders, so `awkward::BuilderOptions` or `awkward::LayoutBuilder::Options`.

Reply:
It's actually a tuple of options with an `awkward::BuilderOptions` specialisation that is a `awkward::Options<int64_t, double>`

Reply:
eventually we can type alias other specialisations and the getters

Reply:
done